packages/react-sdk/src/tree/create-elements-tree.tsx

```diff
@@ -8,7 +8,11 @@ import { SessionStoragePolyfill } from "./session-storage-polyfill";
 
 export type ChildrenUpdates = Array<
   | string
-  | { id: Instance["id"]; component: Instance["component"]; text: string }
+  | {
+      id: null | Instance["id"];
```

Comment:
if its optional, lets make it optional/undefined, not null. I am trying to fight null whenever I can, only using it where I can't avoid it

Reply:
null is not that bad, instead of `undefined`, it shows that the developer explicitly set it as null, or the developer must explicitly set it as null.

`undefined` or optional don't give any information - is it mistake or what

Reply:
Usually if its not set, its not set. You shouldn't care why. Unless you can explain how this information is relevant here, lets use undefined. Generally undefined works with default arguments etc. and nicer to type

Reply:
I meant if its undefined. Unset in objects is a different thing, property still exists here.

Reply:
This is btw mainly how I differentiate if it was set to undefined vs it wasn't set at all. `property in object`

Reply:
https://www.typescriptlang.org/tsconfig#exactOptionalPropertyTypes must be set if not already

Reply:
I probably do explicit delete/unset semantics when I need something to be gone, on the level of an action/event, not on a level of a value that has been set to null

Reply:
in this case I still don't understand if we need to know that something was "deleted"

Reply:
as opposed to being simply optional